File: Programs/Best_Time_to_Buy_and_Sell_Stock_II.cpp
#include<bits/stdc++.h>
using namespace std;

/* LeetCode */
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n= prices.size();
        int profit= 0;

        for(int i=1; i<n; i++) {
            if(prices[i-1] < prices[i])
                profit += prices[i] - prices[i-1];
        }

        return profit;
    }
};


/* CodingNinjas */
int bestTimeToBuyAndSellStockII(vector<int>& prices) {
    int n= prices.size();
    int profit= 0;
    for(int i=1; i<n; i++) {
        if(prices[i-1] < prices[i])
            profit += prices[i] - prices[i-1];
    }
    return profit;
}


/* GekksForGeeks */
class Solution {
public:
    int maximumProfit(vector<int>& prices) {
        int n= prices.size();
        int profit= 0;
        
        for(int i=1; i<n; i++) {
            if(prices[i-1] < prices[i])
                profit += prices[i] - prices[i-1];
        }
        
        return profit;
    }
};

File: Programs/Best_Time_to_Buy_and_Sell_Stock_II.py
""" LeetCode """
class Solution:
    def maxProfit(self, prices: list[int]) -> int:
        profit= 0
        for i in range(1, len(prices)):
            if prices[i-1] < prices[i]:
                profit += prices[i] - prices[i-1]
        return profit


""" CodingNinjas """
def bestTimeToBuyAndSellStockII(prices: list[int]) -> int:
    profit= 0
    for i in range(1, len(prices)):
        if prices[i-1] < prices[i]:
            profit += prices[i] - prices[i-1]
    return profit


""" GeeksForGeeks """
class Solution:
    def maximumProfit(self, prices) -> int:
        profit= 0
        for i in range(1, len(prices)):
            if prices[i-1] < prices[i]:
                profit += prices[i] - prices[i-1]
        return profit

